src/browser/web-viewer.js

```diff
@@ -206,7 +212,9 @@ define((require, exports, module) => {
   WebViewer.Deck = Component('WebViewerDeck', (options, {onOpen, onClose, edit}) => {
     const {items, In} = options;
     return DOM.div(options, items.sortBy(id).map(item => WebViewer({
-      key: item.get('id'),
+      // Change key depending on privileges,
+      // forcing a re-creation of the iframe if privilege level change
+      key: item.get('id') + (url.isPrivileged(item.get('uri')) ? 'PRIVILEGED' : 'NOTPRIVILEGED'),
```

Comment:
This works, although I don't think forcing re-rendering as such is very clean & not easy to scale if number of properties will grow.

The real problem here is that there are non-standard attribute changing which at the DOM level has no effect. I think it would de best to fix it actually on the platform level. In the meantime I think it would be best to change our `Element` implementation to handle such cases better, in fact our [`BeforeAppendAttribute`](https://github.com/mozilla/browser.html/blob/master/src/browser/element.js#L89-L107) is kind of does that already so what I'd do is:

1. Change it's name be something more adequate like `RemountAttribute` or something like that.
2. Add a `write` method to it's implementation that would [call a super method](https://github.com/mozilla/browser.html/blob/master/src/browser/element.js#L128-L136) and then re-inject node to force it to be picked up:

```js
BeforeAppendAttribute.prototype = {
    constructor: BeforeAppendAttribute,
    mount(node, value) {
      node.setAttribute(this.name, value);
    },
   write(node, present, past) {
     Attribute.prototype.write.call(node, present, past)
     if (present !== past) {
       node.parentNode.replaceChild(node, node)
     }
   }
  };
```

This way we'll force node re-creation for attributes that require that. That way user of component like `IFrame` does not need to know intimate details of `IFrame` implementation to make use of them.